Code/Framework/AzCore/Tests/Settings/SettingsRegistryMergeUtilsTests.cpp

```diff
@@ -751,4 +752,277 @@ tags=tools,renderer,metal)"
+            SettingsRegistryFindEngineRootParams{
+                // engine.json files
+                {
+                   R"({ "engine_name": "o3de", "version": "1.2.3"})",
+                   R"({ "engine_name": "o3de", "version": "2.3.4"})",
+                },
+
+                // project/project.json
+                R"({ "project_name": "TestProject", "engine":"o3de" })",
+
+                // project/user/project.json
+                R"({})",
+
+                1 // expect second engine with higher version number
+            },
+
+            // Fails to find engine with name that isn't registered
+            SettingsRegistryFindEngineRootParams{
+                // engine.json files
+                {
+                   R"({ "engine_name": "o3de1", "version": "1.0.0"})",
+                   R"({ "engine_name": "o3de2", "version": "1.2.3"})",
+                },
+
+                // project/project.json
+                R"({ "project_name": "TestProject", "engine":"o3de-not-found"  })",
+
+                // project/user/project.json
+                R"({})",
+
+                -1 // not found
+            },
+
+            // Fails to find engine with version that isn't registered 
+            SettingsRegistryFindEngineRootParams{
+                // engine.json files
+                {
+                   R"({ "engine_name": "o3de1", "version": "1.0.0"})",
+                   R"({ "engine_name": "o3de2", "version": "1.2.3"})",
+                },
+
+                // project/project.json
+                R"({ "project_name": "TestProject", "engine":"o3de==1.1.1"  })",
+
+                // project/user/project.json
+                R"({})",
+
+                -1 // not found
+            },
+
+            // Selects engine that has a legacy version field
+            SettingsRegistryFindEngineRootParams{
+                // engine.json files
+                {
+                   R"({ "engine_name": "o3de", "O3DEVersion": "0.1.0.0"})",
+                   R"({ "engine_name": "o3de-new", "O3DEVersion": "0.1.0.0", "version": "1.2.3"})",
+                },
+
+                // project/project.json
+                R"({ "project_name": "TestProject", "engine":"o3de"  })",
+
+                // project/user/project.json
+                R"({})",
+
+                0 // o3de 
+            },
+
+            // Selects first engine when multiple engines found with ambiguous project engine
+            SettingsRegistryFindEngineRootParams{
+                // engine.json files
+                {
+                   R"({ "engine_name": "o3de", "O3DEVersion": "0.1.0.0"})",
+                   R"({ "engine_name": "o3de", "O3DEVersion": "0.1.0.0"})",
+                },
+
+                // project/project.json
+                R"({ "project_name": "TestProject", "engine":"o3de"  })",
+
+                // project/user/project.json
+                R"({})",
+
+                0 // first engine
+            },
```

Comment:
Can we add a warning when an engine with the same name and version is a candidate?